### PR TITLE
console.lua: add more completers

### PR DESCRIPTION
**Pull request by guidocella**
This adds some more completers to console.lua.

Question: do we want to complete `del` with the list option verbs even though it's deprecated or is it one of these things that no one is going to remove without wm4?

### CODE REVIEW

File: player/lua/console.lua [hunk 1]
```diff
@@ -652,6 +659,50 @@ local function property_list()
...
+end
+
+local function list_option_list()
+    local options = {}
+
+    -- Don't log errors for renamed and removed properties.
+    -- (Just mp.enable_messages('fatal') still logs them to the terminal.)
+    local msg_level_backup = mp.get_property('msg-level')
+    mp.set_property('msg-level', msg_level_backup == '' and 'cplayer=no'
+                                 or msg_level_backup .. ',cplayer=no')
```
Comment (Dudemanguy) [inline player/lua/console.lua:679]: Shouldn't `msg_level_backup..',cplayer=no'` always work?

**Review by christoph-heinrich — APPROVED** [review]
Works well and there's nothing objectionable in the code.
I'm not convinced there is a need for caching the file commands, but I'm also not against it.

Paths with spaces get completed without quotes even though mpv then doesn't parse them correctly. Technically we could not complete paths with spaces if there aren't any quotes, but I'm not sure that's even desirable.
As far as I'm concerned it can also stay as is.

**Comment by christoph-heinrich** [comment]
Forgot about the `del` thing.

We shouldn't encourage using deprecated behavior. Ideally we'd even remove it altogether, it's been deprecated for long enough.

**Comment by Dudemanguy** [comment]
>Paths with spaces get completed without quotes even though mpv then doesn't parse them correctly.

Yeah noticed this too. I think it's just that `mp.command` isn't smart enough to handle this. As far as I'm concerned, it's not related directly to this PR. The fix would be to make the command execution smarter not try to automatically quote the file path or something like that.

**Review by Dudemanguy — APPROVED** [review]
Just needs a rebase.

**Comment by guidocella** [comment]
> I'm not convinced there is a need for caching the file commands, but I'm also not against it.

Me neither but it is just 4 lines.

> Paths with spaces get completed without quotes even though mpv then doesn't parse them correctly. Technically we could not complete paths with spaces if there aren't any quotes, but I'm not sure that's even desirable. As far as I'm concerned it can also stay as is.

I can maybe add a feature to somehow transform the input line later and add quotes if missing as I would also like it to expand `~/`.

> Yeah noticed this too. I think it's just that mp.command isn't smart enough to handle this. As far as I'm concerned, it's not related directly to this PR. The fix would be to make the command execution smarter not try to automatically quote the file path or something like that.

Not possible because commands have other arguments after the file.

**Comment by Dudemanguy** [comment]
>Not possible because commands have other arguments after the file.

I was thinking like incorporating mp.commandv or so. Can be for later of course.

**Pull request merged**